test/unit/acknowledgement/process-invalid.test.js

```diff
@@ -0,0 +1,242 @@
+  })
+
+  test('should call mockTransaction.commit once when an unsuccessful ack object is given', async () => {
+    await processInvalid(schemeId, paymentRequestId, mockFRN, mockAcknowledgementError)
+    expect(mockTransaction.commit).toHaveBeenCalledTimes(1)
+  })
+
+  test('should not call mockTransaction.rollback when an unsuccessful ack object is given and nothing throws', async () => {
+    await processInvalid(schemeId, paymentRequestId, mockFRN, mockAcknowledgementError)
+    expect(mockTransaction.rollback).not.toHaveBeenCalled()
+  })
+
+  test('should call mockTransaction.rollback when resetPaymentRequestById throws', async () => {
+    resetPaymentRequestById.mockRejectedValue(new Error())
+    try { await processInvalid(schemeId, paymentRequestId, mockFRN, mockAcknowledgementError) } catch { }
+    expect(mockTransaction.rollback).toHaveBeenCalled()
+  })
+
+  test('should call mockTransaction.rollback once when resetPaymentRequestById throws', async () => {
+    resetPaymentRequestById.mockRejectedValue(new Error())
+    try { await processInvalid(schemeId, paymentRequestId, mockFRN, mockAcknowledgementError) } catch { }
+    expect(mockTransaction.rollback).toHaveBeenCalledTimes(1)
+  })
+
+  test('should call mockTransaction.rollback when getHoldCategoryId throws', async () => {
+    getHoldCategoryId.mockRejectedValue(new Error())
+    try { await processInvalid(schemeId, paymentRequestId, mockFRN, mockAcknowledgementError) } catch { }
+    expect(mockTransaction.rollback).toHaveBeenCalled()
+  })
+
+  test('should call mockTransaction.rollback once when getHoldCategoryId throws', async () => {
+    getHoldCategoryId.mockRejectedValue(new Error())
+    try { await processInvalid(schemeId, paymentRequestId, mockFRN, mockAcknowledgementError) } catch { }
+    expect(mockTransaction.rollback).toHaveBeenCalledTimes(1)
+  })
+
+  test('should call mockTransaction.rollback when holdAndReschedule throws', async () => {
+    holdAndReschedule.mockRejectedValue(new Error())
+    try { await processInvalid(schemeId, paymentRequestId, mockFRN, mockAcknowledgementError) } catch { }
+    expect(mockTransaction.rollback).toHaveBeenCalled()
+  })
+
+  test('should call mockTransaction.rollback once when holdAndReschedule throws', async () => {
+    holdAndReschedule.mockRejectedValue(new Error())
+    try { await processInvalid(schemeId, paymentRequestId, mockFRN, mockAcknowledgementError) } catch { }
+    expect(mockTransaction.rollback).toHaveBeenCalledTimes(1)
+  })
+
+  test('should call mockTransaction.rollback when mockTransaction.commit throws', async () => {
+    mockTransaction.commit.mockRejectedValue(new Error())
+    try { await processInvalid(schemeId, paymentRequestId, mockFRN, mockAcknowledgementError) } catch { }
+    expect(mockTransaction.rollback).toHaveBeenCalled()
+  })
+
+  test('should call mockTransaction.rollback once when mockTransaction.commit throws', async () => {
```

Comment:
Would add tests that `processInvalid` throws when any of these functions throw as per line 23 in the function itself

Reply:
Do you mean a different line, as line 23 has:

jest.mock('../../../app/event')
const mockEvent = require('../../../app/event') ?